gluten-core/src/main/scala/org/apache/gluten/extension/injector/RuleInjector.scala

```diff
@@ -19,12 +19,13 @@ package org.apache.gluten.extension.injector
 import org.apache.spark.sql.SparkSessionExtensions
 
 /** Injector used to inject query planner rules into Spark and Gluten. */
-class RuleInjector {
-  val spark: SparkInjector = new SparkInjector()
+class RuleInjector(extensions: SparkSessionExtensions) {
+  val spark: SparkInjector = new SparkInjector(extensions)
   val gluten: GlutenInjector = new GlutenInjector()
 
   private[extension] def inject(extensions: SparkSessionExtensions): Unit = {
-    spark.inject(extensions)
+    // The regular Spark rules already injected with the `injectRules` of `RuleApi` directly.
+    // Only inject the Spark columnar rule here.
     gluten.inject(extensions)
   }
```

Comment:
Both constructor and method `#inject` accepts a `extensions: SparkSessionExtensions`. Can we remove the one in `#inject`?

Reply:
updated.